open-morello-serial-ports.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/python3
+import argparse
+import logging
+import os
+import shlex
+import shutil
+import sys
+
+from enum import Enum
+
+from serial.tools.list_ports import comports  # noqa: E402
+from serial.tools.list_ports_common import ListPortInfo  # noqa: E402
+
+
+def find_morello_board_ttys(pretend: bool) -> "list[ListPortInfo]":
+    # find the serial port:
+    # Run `ioreg -p IOUSB -l -w 0` on macOS to find the right VID/PID
+    expected_vendor_id = 0x0403
+    expected_product_id = 0x6011
+    ttys = []
+    for portinfo in comports(include_links=True):
+        logging.debug("Checking %s", portinfo)
+        assert isinstance(portinfo, ListPortInfo)
+        if portinfo.pid == expected_product_id and portinfo.vid == expected_vendor_id:
+            ttys.append(portinfo)
+    if len(ttys) != 8:
+        if pretend:
+            return [ListPortInfo("/dev/fakeTTY")] * 8
+        raise ValueError("Could not find 8 USB TTYs with VID", hex(expected_vendor_id), "PID", hex(expected_product_id))
+    # Sort by location, then device name, since B0-3 actually are ports 0-3 and A0-3 are 4-7
+    ttys = list(sorted(ttys, key=lambda x: (x.location, x.device)))
+    return ttys
+
+
+# List of Morello board uarts, see
+# https://git.morello-project.org/morello/docs/-/blob/morello/mainline/user-guide.rst#setting-up-the-morello-board
+# FIXME: documentation says AP2 is +7, but it appears to be +5 instead
```

Comment:
Hm.  Oddly, I've been unable to find AP2 on one of the boards I'm testing on.  I assume I'm doing it wrong; would you mind sharing your test methodology here?

Reply:
I am only assuming this because that is the UART that prints the EL3 panic when I try to reboot.